bascule/basculehttp/tokenFactory.go

```diff
@@ -45,3 +59,68 @@ func (btf BasicTokenFactory) ParseAndValidate(ctx context.Context, _ *http.Reque
 	// failed authentication
 	return nil, errors.New("TODO: Enrich this error with information")
 }
+
+type BearerTokenFactory struct {
+	DefaultKeyId  string
+	Resolver      key.Resolver
+	Parser        bascule.JWSParser
+	JWTValidators []*jwt.Validator
+}
+
+func (btf BearerTokenFactory) ParseAndValidate(ctx context.Context, request *http.Request, auth bascule.Authorization, value string) (bascule.Token, error) {
+	decoded, err := base64.StdEncoding.DecodeString(value)
+	if err != nil {
+		return nil, err
+	}
+
+	jwsToken, err := btf.Parser.ParseJWS(decoded)
+	if err != nil {
+		return nil, err
+	}
+
+	protected := jwsToken.Protected()
+	if len(protected) == 0 {
+		return nil, ErrorNoProtectedHeader
+	}
+
+	alg, _ := protected.Get("alg").(string)
+	signingMethod := jws.GetSigningMethod(alg)
+	if signingMethod == nil {
+		return nil, ErrorNoSigningMethod
+	}
+
+	keyID, _ := protected.Get("kid").(string)
+	if len(keyID) == 0 {
+		keyID = btf.DefaultKeyId
+	}
+
+	pair, err := btf.Resolver.ResolveKey(keyID)
+	if err != nil {
+		return nil, err
+	}
+
+	// validate the signature
+	if len(btf.JWTValidators) > 0 {
+		// all JWS implementations also implement jwt.JWT
+		err = jwsToken.(jwt.JWT).Validate(pair.Public(), signingMethod, btf.JWTValidators...)
+	} else {
+		err = jwsToken.Verify(pair.Public(), signingMethod)
+	}
+
+	if err != nil {
+		// todo: add metrics to log the type of verification error
+		return nil, err
+	}
+
+	payload, ok := jwsToken.Payload().(bascule.Attributes)
```

Comment:
The payload won't be of type `bascule.Attributes`.  You'll need to convert it.  The easiest way is probably an explicit conversion:

```
payload := bascule.Attributes(jwsToken.Payload())
```

The difference is that you aren't asking if Payload() is an Attributes.  You're forcing it to be an Attributes.

Reply:
fixed in commit e7bd07e